#########################Automated tests trace#########################

self = <MEGS.Tests.ListBased.test_list_based_action.TestListBased object at 0x041D3C30>
list_entity_test_criteria = <PyTest.CustomFixtures.custom_fixtures.ListEntityTestCriteria object at 0x035612F0>
search_api = <Search.Implementation.Contract.search.SearchAPI object at 0x041D3A50>

    @pytest.mark.crud
    def test_list_delete_member(self, list_entity_test_criteria, search_api):
        if isinstance(list_entity_test_criteria, ListEntityTestCriteria):
            list_base = list_entity_test_criteria.api_object
            list_base.es_client = self.es_client
            list_logical_name = list_entity_test_criteria.entity_name
            member_type = list_entity_test_criteria.member_type
            list_member_type_logical_name = list_base.list_type_member_mapping(member_type)
            _primary_key = self.es_client.edge_metadata[list_logical_name].CacheSchema.get_fields(flags=["IsKey"])[0]["PropertyName"]
    
            ###Create new marketing list###
            create_response = list_base.create_entity(params={"member_type": member_type})
    
            #Check whether request has been performed successfully
            self.assertHttpResponseCode(http_response=create_response)
    
            #Create model
            create_resp_model = self.create_entity_model(
                response=create_response[1],
                primary_key_field=_primary_key,
                search_api=search_api)
    
            #check list in index
            self.assertEntityInIndex(
                search_api=search_api,
                entity_type=list_logical_name,
                entityid=create_resp_model.id
            )
    
            #get random entity from ES by list_member_type_logical_name to add to list
            random_entity_id = self.es_client.get_entity_by_type(
                entity_type=list_member_type_logical_name
>           )[0]
E           IndexError: list index out of range

C:\EdgeDeployment\FunctionalTests\src\MEGS\Tests\ListBased\test_list_based_action.py:155: IndexError
-------------------------------- Captured log ---------------------------------
requests_api.py             76 INFO     Sending HTTP request...
requests_api.py             83 DEBUG    Method type: GET
requests_api.py             94 DEBUG    Url: http://localhost/xre2113qa.nightly/api/authentication
requests_api.py             95 DEBUG    Headers: {'Authorization-Token': u'6569219054d540fb97925f7d0b93ffb3', 'Accept-Language': 'en-US,en;q=0.8', 'Content-Type': 'application/json', 'Accept-Encoding': 'deflate, gzip'}
connectionpool.py          176 INFO     Starting new HTTP connection (1): localhost
connectionpool.py          344 DEBUG    "GET /xre2113qa.nightly/api/authentication HTTP/1.1" 200 2097
requests_api.py            140 INFO     Response recieved...
requests_api.py            141 INFO     Response status: 200
requests_api.py            142 INFO     Response reason: OK
requests_api.py            143 INFO     Response time: 0:00:00.750000
EntityBase.py              272 DEBUG    Creating request body...
connectionpool.py          296 DEBUG    "GET /xre2113qa.nightly.settings/ValueWithVersion%601/systemuser HTTP/1.1" 200 3980
connectionpool.py          296 DEBUG    "GET /xre2113qa.nightly.settings/ValueWithVersion%601/activitypointer HTTP/1.1" 200 9299
connectionpool.py          296 DEBUG    "GET /xre2113qa.nightly.settings/ValueWithVersion%601/systemuser_1 HTTP/1.1" 404 104
elasticsearch_client.py     60 WARNING  ValueWithVersion`1 entity has not been found in index xre2113qa.nightly.settings
connectionpool.py          296 DEBUG    "GET /xre2113qa.nightly/systemuser_1/_count HTTP/1.1" 200 60
connectionpool.py          296 DEBUG    "GET /xre2113qa.nightly/_search HTTP/1.1" 200 935
connectionpool.py          296 DEBUG    "GET /xre2113qa.nightly.settings/ValueWithVersion%601/businessunit HTTP/1.1" 404 104
elasticsearch_client.py     60 WARNING  ValueWithVersion`1 entity has not been found in index xre2113qa.nightly.settings
connectionpool.py          296 DEBUG    "GET /xre2113qa.nightly.settings/ValueWithVersion%601/activitypointer HTTP/1.1" 200 9299
connectionpool.py          296 DEBUG    "GET /xre2113qa.nightly.settings/ValueWithVersion%601/businessunit HTTP/1.1" 404 104
elasticsearch_client.py     60 WARNING  ValueWithVersion`1 entity has not been found in index xre2113qa.nightly.settings
connectionpool.py          296 DEBUG    "GET /xre2113qa.nightly/businessunit/_count HTTP/1.1" 200 59
connectionpool.py          296 DEBUG    "GET /xre2113qa.nightly/_search HTTP/1.1" 200 123
connectionpool.py          296 DEBUG    "GET /xre2113qa.nightly.settings/ValueWithVersion%601/team HTTP/1.1" 404 96
elasticsearch_client.py     60 WARNING  ValueWithVersion`1 entity has not been found in index xre2113qa.nightly.settings
connectionpool.py          296 DEBUG    "GET /xre2113qa.nightly.settings/ValueWithVersion%601/activitypointer HTTP/1.1" 200 9299
connectionpool.py          296 DEBUG    "GET /xre2113qa.nightly.settings/ValueWithVersion%601/team HTTP/1.1" 404 96
elasticsearch_client.py     60 WARNING  ValueWithVersion`1 entity has not been found in index xre2113qa.nightly.settings
connectionpool.py          296 DEBUG    "GET /xre2113qa.nightly/team/_count HTTP/1.1" 200 59
connectionpool.py          296 DEBUG    "GET /xre2113qa.nightly/_search HTTP/1.1" 200 123
EntityBase.py              383 DEBUG    Request body created:
EntityBase.py              384 DEBUG    {"awx_membertypecode": "Availabilities", "awx_version": 101, "modifiedonbehalfby": {"logicalname": "systemuser", "id": "e8b9896a-be56-e311-9314-00155d280475"}, "type": "awx_adhoclist", "timezoneruleversionnumber": 122}
requests_api.py             76 INFO     Sending HTTP request...
requests_api.py             83 DEBUG    Method type: POST
requests_api.py             94 DEBUG    Url: http://localhost/xre2113qa.nightly/api/adhoclist//
requests_api.py             95 DEBUG    Headers: {'Authorization-Token': u'6569219054d540fb97925f7d0b93ffb3', 'Accept-Language': 'en-US,en;q=0.8', 'Content-Type': 'application/json', 'Accept-Encoding': 'deflate, gzip'}
requests_api.py            104 DEBUG    Request body: {"awx_membertypecode": "Deals", "awx_version": 101, "modifiedonbehalfby": {"logicalname": "systemuser", "id": "e8b9896a-be56-e311-9314-00155d280475"}, "type": "awx_adhoclist", "timezoneruleversionnumber": 122}
connectionpool.py          176 INFO     Starting new HTTP connection (1): localhost
connectionpool.py          344 DEBUG    "POST /xre2113qa.nightly/api/adhoclist// HTTP/1.1" 200 1398
requests_api.py            140 INFO     Response recieved...
requests_api.py            141 INFO     Response status: 200
requests_api.py            142 INFO     Response reason: OK
requests_api.py            143 INFO     Response time: 0:00:01.330000
requests_api.py             76 INFO     Sending HTTP request...
requests_api.py             83 DEBUG    Method type: POST
requests_api.py             94 DEBUG    Url: http://localhost/xre2113qa.nightly/api/search/awx_adhoclist
requests_api.py             95 DEBUG    Headers: {'Authorization-Token': u'6569219054d540fb97925f7d0b93ffb3', 'Accept-Language': 'en-US,en;q=0.8', 'Content-Type': 'application/json', 'Accept-Encoding': 'deflate, gzip'}
requests_api.py            104 DEBUG    Request body: {"ResultsFormat": null, "From": 0, "SelectedDateTimeFacets": null, "Relations": [], "PinTo": {"lat": -27.47018, "lon": 153.02911}, "NumericFacetFilters": [], "OrderBy": [{"Field": "_score", "IsDescending": "true"}], "GeoPolygonFilters": [], "TermFacetFilters": [], "FieldFilters": [], "GeoDistanceFacetFilters": [], "ScriptFilters": [], "Ids": ["9dcdd5e6-7998-e311-aa29-00155d28de35"], "SelectedNumericFacets": null, "RelationCounts": [], "Options": null, "SelectedTermFacets": null, "DateTimeFacetFilters": [], "TermFacetSize": 1, "SelectedGeoFacets": null, "Query": "*", "Size": 1}
connectionpool.py          176 INFO     Starting new HTTP connection (1): localhost
connectionpool.py          344 DEBUG    "POST /xre2113qa.nightly/api/search/awx_adhoclist HTTP/1.1" 200 2552
requests_api.py            140 INFO     Response recieved...
requests_api.py            141 INFO     Response status: 200
requests_api.py            142 INFO     Response reason: OK
requests_api.py            143 INFO     Response time: 0:00:00.952000
IncrementalUpdateAssertMixin.py  136 ERROR    Unable to get entity Index Version from index. entity-type=awx_adhoclist|entity_id=9dcdd5e6-7998-e311-aa29-00155d28de35 
IncrementalUpdateAssertMixin.py  138 ERROR    'IndexVersion'
Traceback (most recent call last):
  File "C:\EdgeDeployment\FunctionalTests\src\MEGS\Implementation\Assertions\IncrementalUpdateAssertMixin.py", line 132, in check_entity_index_version
    index_version = obj['Results'][0]['IndexVersion']
KeyError: 'IndexVersion'
IncrementalUpdateAssertMixin.py   71 DEBUG    #######################
IncrementalUpdateAssertMixin.py   72 DEBUG    Search awx_adhoclist in Index with Id - 9dcdd5e6-7998-e311-aa29-00155d28de35 -
requests_api.py             76 INFO     Sending HTTP request...
requests_api.py             83 DEBUG    Method type: POST
requests_api.py             94 DEBUG    Url: http://localhost/xre2113qa.nightly/api/search/awx_adhoclist
requests_api.py             95 DEBUG    Headers: {'Authorization-Token': u'6569219054d540fb97925f7d0b93ffb3', 'Accept-Language': 'en-US,en;q=0.8', 'Content-Type': 'application/json', 'Accept-Encoding': 'deflate, gzip'}
requests_api.py            104 DEBUG    Request body: {"ResultsFormat": null, "From": 0, "SelectedDateTimeFacets": null, "Relations": [], "PinTo": {"lat": -27.47018, "lon": 153.02911}, "NumericFacetFilters": [], "OrderBy": [{"Field": "_score", "IsDescending": "true"}], "GeoPolygonFilters": [], "TermFacetFilters": [], "FieldFilters": [], "GeoDistanceFacetFilters": [], "ScriptFilters": [], "Ids": ["9dcdd5e6-7998-e311-aa29-00155d28de35"], "SelectedNumericFacets": null, "RelationCounts": [], "Options": null, "SelectedTermFacets": null, "DateTimeFacetFilters": [], "TermFacetSize": 1, "SelectedGeoFacets": null, "Query": "*", "Size": 1}
connectionpool.py          176 INFO     Starting new HTTP connection (1): localhost
connectionpool.py          344 DEBUG    "POST /xre2113qa.nightly/api/search/awx_adhoclist HTTP/1.1" 200 2552
requests_api.py            140 INFO     Response recieved...
requests_api.py            141 INFO     Response status: 200
requests_api.py            142 INFO     Response reason: OK
requests_api.py            143 INFO     Response time: 0:00:00.580000
IncrementalUpdateAssertMixin.py  160 DEBUG    Search Result for Id - 9dcdd5e6-7998-e311-aa29-00155d28de35 -:
IncrementalUpdateAssertMixin.py  161 DEBUG    {"modifiedon": "2014-02-18T08:51:51Z", "awx_membertypecode": "Deals", "awx_adhoclistid": "9dcdd5e6-7998-e311-aa29-00155d28de35", "awx_name": null, "versionnumber": 2082979, "statecode": "Active", "statuscode": "Active"}
IncrementalUpdateAssertMixin.py   92 DEBUG    Entity with id - 9dcdd5e6-7998-e311-aa29-00155d28de35 - exist in search: True
IncrementalUpdateAssertMixin.py   93 DEBUG    #######################
connectionpool.py          296 DEBUG    "GET /xre2113qa.nightly.settings/ValueWithVersion%601 HTTP/1.1" 400 92
elasticsearch_client.py     60 WARNING  ValueWithVersion`1 entity has not been found in index xre2113qa.nightly.settings
connectionpool.py          296 DEBUG    "GET /xre2113qa.nightly.settings/ValueWithVersion%601/activitypointer HTTP/1.1" 200 9299
connectionpool.py          296 DEBUG    "GET /xre2113qa.nightly.settings/ValueWithVersion%601 HTTP/1.1" 400 92
elasticsearch_client.py     60 WARNING  ValueWithVersion`1 entity has not been found in index xre2113qa.nightly.settings
connectionpool.py          296 DEBUG    "GET /xre2113qa.nightly/_count HTTP/1.1" 200 62
#########################Server error trace#########################
{"index":[],"megs":[{"TimeStamp":"2014-02-18T08:50:54.744685Z","Message":"FaultException: ","ErrorLevel":"ERROR","Longdate":"2014-02-18 02:50:54.6216","ThreadId":"80 ","Logger":"AscendixRE.MEGS.ApiControllers.AdHocListController","Exception":"System.ServiceModel.FaultException`1[Microsoft.Xrm.Sdk.OrganizationServiceFault]: A validation error occurred.  The value of 'awx_membertypecode' on record of type 'awx_adhoclist' is outside the valid range. (Fault Detail is equal to Microsoft.Xrm.Sdk.OrganizationServiceFault)."},{"TimeStamp":"2014-02-18T08:50:54.7925335Z","Message":"Inner Fault Message: A validation error occurred.  The value of 'awx_membertypecode' on record of type 'awx_adhoclist' is outside the valid range.","ErrorLevel":"ERROR","Longdate":"2014-02-18 02:50:54.6216","ThreadId":"80 ","Logger":"AscendixRE.MEGS.ApiControllers.AdHocListController","Exception":""},{"TimeStamp":"2014-02-18T08:50:54.9136195Z","Message":"Inner Fault Trace Text: \r\n[Microsoft.Crm.ObjectModel: Microsoft.Crm.ObjectModel.TargetAttributeValidationPlugin]\r\n[c754fdb4-ef2f-4f2c-b49f-186214ab4dcd: TargetAttributeValidationPlugin]\r\n\r\n","ErrorLevel":"ERROR","Longdate":"2014-02-18 02:50:54.6216","ThreadId":"80 ","Logger":"AscendixRE.MEGS.ApiControllers.AdHocListController","Exception":""},{"TimeStamp":"2014-02-18T08:50:55.0190815Z","Message":"Exception: ","ErrorLevel":"ERROR","Longdate":"2014-02-18 02:50:54.6216","ThreadId":"80 ","Logger":"AscendixRE.MEGS.ApiControllers.AdHocListController","Exception":"System.ArgumentException: Bad input data: A validation error occurred.  The value of 'awx_membertypecode' on record of type 'awx_adhoclist' is outside the valid range.\r\n   at AscendixRE.MEGS.ApiControllers.BaseControllers.EntityControllerBase.CheckSenderFault(FaultException ex)\r\n   at AscendixRE.MEGS.ApiControllers.BaseControllers.EntityControllerBase.CreateEntity(Entity sdkEntity)\r\n   at AscendixRE.MEGS.ApiControllers.BaseControllers.EntityControllerBase.Post(Object entity)\r\n   at System.Web.Http.Controllers.ReflectedHttpActionDescriptor.ActionExecutor.<>c__DisplayClass13.<GetExecutor>b__c(Object instance, Object[] methodParameters)\r\n   at System.Web.Http.Controllers.ReflectedHttpActionDescriptor.ActionExecutor.Execute(Object instance, Object[] arguments)\r\n   at System.Threading.Tasks.TaskHelpers.RunSynchronously[TResult](Func`1 func, CancellationToken cancellationToken)"}]}
